curvefs/src/client/dir_buffer.cpp

```diff
@@ -28,16 +28,59 @@ namespace client {
 
 
 uint32_t DirBuffer::DirBufferNew() {
-    return 0;
+    uint32_t dindex = -1;
+    {
+        curve::common::LockGuard lg(indexMtx_);
+        if (recycleIndex_.empty()) {
+            dindex = index_++;
+        } else {
+            dindex = recycleIndex_.front();
+            recycleIndex_.pop_front();
+        }
+    }
+    curve::common::WriteLockGuard wlg(bufferMtx_);
+    DirBufferHead *head = new DirBufferHead();
```

Comment:
How to deal with the old dir buffer head and the buffer in old dir buffer head

Reply:
the buffer should be release by forget